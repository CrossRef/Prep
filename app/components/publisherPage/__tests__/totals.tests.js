import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon'

import Totals from '../components/totals'




//Dummy Data
const totals = {
  "journal-articles":12345,
  "books":1234,
  "conference-papers":123,
  "reports":123,
  "datasets":0,
  "dissertations":0,
  "posted-content":0,
  "standards":0
}


//Render and begin tests
const shallowComponent = shallow(<Totals totals={totals}/>)


test('shallow snapshot', () => {
  expect(toJson(shallowComponent)).toMatchSnapshot()
})