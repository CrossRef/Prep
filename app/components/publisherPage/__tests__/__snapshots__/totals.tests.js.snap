// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow snapshot 1`] = `
<div
  className="publisherTotals"
>
  <div
    className="totalTooltip"
  >
    <p
      key="journal-articles"
    >
      Journal articles: 12,345
    </p>
    <p
      key="books"
    >
      Books: 1,234
    </p>
    <p
      key="conference-papers"
    >
      Conference papers: 123
    </p>
    <p
      key="reports"
    >
      Reports: 123
    </p>
  </div>
  <div
    className="tooltipHoverArea"
  >
    <p
      key="journal-articles"
    >
      Journal articles: 12,345
    </p>
    <p
      key="books"
    >
      Books: 1,234
    </p>
    <p
      key="conference-papers"
    >
      Conference papers: 123
    </p>
    <p
      key="reports"
    >
      Reports: 123
    </p>
  </div>
  <p
    className="totalNumber"
  >
    13,825
  </p>
  <p
    className="totalText"
  >
    Total registered content items
  </p>
</div>
`;
